### ISSUE TITLE
CUDA 10 build break

### ISSUE THREAD

**Issue by ngeorgis** [facebookresearch/wav2letter#161]
- Successful CPU build.
- Have not tried to build with recommended CUDA 9.2
- Build breaks with CUDA 10 -  building native (not docker).

```make
[  0%] Built target warpctc
[  0%] Performing update step for 'CUB'
[  0%] No configure step for 'CUB'
[  1%] No build step for 'CUB'
[  1%] No install step for 'CUB'
[  1%] Completed 'CUB'
[  1%] Built target CUB
[  1%] Building NVCC (Device) object src/criterion/CMakeFiles/w2lCriterionCudaKernels.dir/backend/cuda/kernels/w2lCriterionCudaKernels_generated_FullConnectionCriterion.cu.o
/home/user/dl/wav2letter/src/criterion/backend/cuda/kernels/FullConnectionCriterion.cu:1:70: fatal error: criterion/backend/cuda/kernels/FullConnectionCriterion.cuh: No such file or directory
compilation terminated.
CMake Error at w2lCriterionCudaKernels_generated_FullConnectionCriterion.cu.o.Release.cmake:215 (message):
  Error generating
  /home/user/dl/wav2letter/build/src/criterion/CMakeFiles/w2lCriterionCudaKernels.dir/backend/cuda/kernels/./w2lCriterionCudaKernels_generated_FullConnectionCriterion.cu.o


src/criterion/CMakeFiles/w2lCriterionCudaKernels.dir/build.make:70: recipe for target 'src/criterion/CMakeFiles/w2lCriterionCudaKernels.dir/backend/cuda/kernels/w2lCriterionCudaKernels_generated_FullConnectionCriterion.cu.o' failed
make[2]: *** [src/criterion/CMakeFiles/w2lCriterionCudaKernels.dir/backend/cuda/kernels/w2lCriterionCudaKernels_generated_FullConnectionCriterion.cu.o] Error 1
CMakeFiles/Makefile2:1134: recipe for target 'src/criterion/CMakeFiles/w2lCriterionCudaKernels.dir/all' failed
make[1]: *** [src/criterion/CMakeFiles/w2lCriterionCudaKernels.dir/all] Error 2
Makefile:140: recipe for target 'all' failed
make: *** [all] Error 2
```

**Comment by ngeorgis** [comment]
**_Update:_**
CUDA10 built fine.
Unit tests pass.
nvprof can see the CUDA kernels.

JFTR the only modification in my system was 

```
git diff
diff --git a/src/criterion/backend/cuda/kernels/FullConnectionCriterion.cu b/src/criterion/backend/cuda/kernels/FullConnectionCriterion.cu
index 00c8f16..969c012 100644
--- a/src/criterion/backend/cuda/kernels/FullConnectionCriterion.cu
+++ b/src/criterion/backend/cuda/kernels/FullConnectionCriterion.cu
@@ -1,4 +1,4 @@
-#include "criterion/backend/cuda/kernels/FullConnectionCriterion.cuh"
+#include "/home/user/dl/wav2letter/src/criterion/backend/cuda/kernels/FullConnectionCriterion.cuh"
 
 #include <cassert>
 #include <cfloat>
diff --git a/src/criterion/backend/cuda/kernels/ViterbiPath.cu b/src/criterion/backend/cuda/kernels/ViterbiPath.cu
index a70ed9d..17d3d90 100644
--- a/src/criterion/backend/cuda/kernels/ViterbiPath.cu
+++ b/src/criterion/backend/cuda/kernels/ViterbiPath.cu
@@ -1,4 +1,4 @@
-#include "criterion/backend/cuda/kernels/ViterbiPath.cuh"
+#include "/home/user/dl/wav2letter/src/criterion/backend/cuda/kernels/ViterbiPath.cuh"
```

**Comment by ngeorgis** [comment]
Training completed.
Testing results look bad - not sure if it my error or a CUDA 10 issue. Will try with CUDA 9 on an older GPU next.

```|P|: habti chef's face expire victo saith acid orde's homra dly keats rayther loads kelsos cahn banzai bunol poca praya olav nat perier kadour ass't nibby eban busily pocus pardy seedy by' larby keys larby ashpot poked crake shots egor peeved amiga blore bardi uyeno tegel keys opium rai ass't by' jib's miot isolde stanly morti lese moe's veille ceorl algoma jarvis's bist rai raree plank badeley 
[sample: 996, WER: 122.222%, LER: 111.61%, slice WER: 159.958%, slice LER: 133.789%, progress: 99.6951%]
|T|: but i stayed that spring and built me a boat 
|P|: habti chef's face expire victo saith acid orde's homra dly keats rayther loads 
[sample: 1467, WER: 130%, LER: 150%, slice WER: 159.913%, slice LER: 133.808%, progress: 100%]
```

**Comment by jacobkahn** [comment]
@ngeorgis — this was an issue that was fixed in 304b62d1f9c4d0c22eec3c4ae1e2689b441f4331, which only affected older versions of CMake. The build issue is unrelated to your model performance. wav2letter hasn't been tested with CUDA 10, so it's possible there are issues training, but I would expect there aren't, and that you'd see similar performance on CUDA 9/with an older GPU.
